include/gmsh.tcc

```diff
@@ -269,16 +271,20 @@ void GMSH<Tdim, Tvertices>::compute_material_points(unsigned ngauss_points) {
       }
 
       //! Make class point and store to material points
-      materialpoints_.at(material_id)
-          ->add_points(std::unique_ptr<Point<Tdim>>(
-              new Point<Tdim>(element->id(), element->id() + last_global_id,
-                              pointsarray, point_volume)));
+      std::shared_ptr<Point<Tdim>> point = std::make_shared<Point<Tdim>>(
+          point_id, point_id + global_id_prev, pointsarray, point_volume);
+      materialpoints_.at(material_id)->add_points(point);
+
+      ++point_id;
     }
+
+    //! Update global_id_prev for next material properties
+    global_id_prev = point_id + 1;
```

Comment:
You are pre-incrementing `point_id` do you need this `+1`?

Reply:
We need it, because it is for another points with different `material_properties`

Reply:
Have you tried printing out the ids? `point_id` is pre-incremented. If this is the `previous_id` then we shouldn't assign it after increment.

Reply:
I think I wasn't communicating this well. Maybe an example would do. 

Remember we might have more than one materials! Say we have two material properties. And each material properties has 5 material points associated with it.

So in the first loop the points start from 0 (`initial_points_id = 0`), say we have point global id of 0 1 2 3 4.
After generating this, I have to change the global id such that it starts with 5 and not 0. Thus I have `initial_points_id = points_id (4 in this case) + 1 = 5`. in the next part where we have `point_id = 0`, the global id will be `global_id = point_id + initial_points_id = 0 + 5 = 5` and so on. Thus we need this variable and it is done so. We can discuss this more tomorrow morning if you are still not convinced

Reply:
I'm happy with the reasoning of a global id. I think the implementation may not be doing what you are expecting it to do. Please print and check if your `id`s are correct. Since you are pre-incrementing you won't need to add `+1` to the `initial_points_id`. Also, please check where the `initial_points_id` is updated?